core/src/network/handler/store.rs

```diff
@@ -32,13 +32,13 @@ pub fn handle_store_dht_meta(dht_meta_data: DhtMetaData, context: Arc<Context>)
                 &serde_json::to_string(&dht_meta_data.content)
                     .expect("dht_meta_data should be EntryWithHader"),
             )
-            .expect("dht_meta_data should be EntryWithHeader");
-            let link_add = match entry_with_header.entry {
-                Entry::LinkAdd(link_add) => link_add,
-                _ => unreachable!(),
-            };
-            let link = link_add.link().clone();
-            let _ = block_on(add_link(&link, &context.clone()));
+            .expect("dht_meta_data should be EntryWithHader");
+            thread::spawn(move || {
```

Comment:
You are spawning a system thread here. This can't be good. The Future should be dispatched to an executor